docs/bare_metal/using_bare_metal.md

```diff
@@ -1,3 +1,72 @@
 # Using the bare metal profile
 
-Content still in PR
+This guide shows how to create a bare metal profile application:
+1. By default, when you build an application binary, the build tool uses the full profile. To use the bare metal profile, you need to set up your application to override this default behaviour.
+1. Bare metal has a minimal set of APIs. You can add additional ones [from the list of supported API](bare_metal.md#features).
+
+## Creating a bare metal application
+
+1. Create a new Mbed application and enter its directory:
+
+    ```
+    mbed new example_app && cd example_app
+    ```
+
+    This contains an empty application with Mbed OS (`mbed-os/`) pre-fetched.
+
+1. Open `mbed_app.json` and replace it with the following content:
+
+    ```json
+    {
+        "requires": ["bare-metal"],
+        "target_overrides": {
+            "*": {
+                "target.c_lib": "small"
+            }
+        }
+    }
+    ```
+    <span class="tips">**Tip:** `"target.c_lib": "small"` enables an optimised version of the C library with lower memory footprint. For more details, see [Using small C libraries in Mbed OS bare metal](c_small_libs.md)</span>.
+
+    Bare metal has a minimal set of default APIs - those that are always available to a bare metal application. You can add other supported APIs if you need the features they enable.
+
+    For a list of default and supported APIs, [please see our full API list](bare_metal.md#features).
+
+1. To add an API - in this example, the `EventQueue` class:
+
+    1. Locate the library you want to use in `mbed-os/`.
+    1. In the library folder, find the library's name in `mbed_lib.json`. You will need it for the next step.
+
+        For example: `events/mbed_lib.json`:
+        ```json
+        {
+            "name": "events",
+            "config": {
+                "present": 1,
+                ...
+            }
+        }
+        ```
+        To continue, go back to the application's root directory.
+
+    1. Open `mbed_app.json` again, and add the library to the `"requires"` field:
+
+        ```json
+        {
+            "requires": [
+                "bare-metal",
+                "events"
```

Comment:
Minor comment: I prefer to see all the required libraries in one line. It helps when using grep.

Reply:
I'll change it to be in one line since there're only two items.